lib/auth/auth.go

```diff
@@ -170,19 +168,19 @@ func (a *AuthServer) GetLocalDomain() (string, error) {
 
 // GenerateHostCert generates host certificate, it takes pkey as a signing
 // private key (host certificate authority)
-func (s *AuthServer) GenerateHostCert(key []byte, hostID, authDomain string, role teleport.Role, ttl time.Duration) ([]byte, error) {
+func (s *AuthServer) GenerateHostCert(key []byte, hostID, authDomain string, roles teleport.Roles, ttl time.Duration) ([]byte, error) {
 	ca, err := s.Trust.GetCertAuthority(services.CertAuthID{
 		Type:       services.HostCA,
 		DomainName: s.DomainName,
 	}, true)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, trace.Errorf("failed to load host CA for '%s': %v", s.DomainName, err)
```

Comment:
The correct way to handle errors is:

- Log the log-level error in-place.
- Form a higher level error and pass it to the caller.

Maybe we should update `trace` package to support this behavior?

In this case I made the change because the function _was_ failing in my real life testing and the low-level message was confusing to a user. The current code delivers superior user experience.
Bubbling up low-level errors up the stack is not helpful to:
a) debug (if you are a developer)
b) change configuration (if you are a user)
Messages like "closed socket" or "bad argument" are only helpful within their context, not outside of it.